fastai/dataset.py

```diff
@@ -47,6 +47,43 @@ def read_dirs(path, folder):
             labels.append(label)
     return filenames, labels, all_labels
 
+def create_sample(path, r):
+    """ Takes a path to a dataset and creates a sample of specified size at <path>_sample
+
+    Parameters:
+    -----------
+    path: dataset path
+    r (float): how many examples to use as a sample, in the range from 0 to 1
+    """
+    sample_path = path + '_sample'
+    shutil.rmtree(sample_path, ignore_errors=True)
+    subdirs = [os.path.split(p)[1] for p in glob(os.path.join(path, '*'))]
+    copy_or_move_with_subdirs(subdirs, path, sample_path, r, move=False)
+
+def create_val(path, r):
+    """ Takes a path to a dataset and creates a validation set of specified size
+
+    Note - this changes the dataset at <path> by moving files to the val set
+
+    Parameters:
+    -----------
+    path: dataset path
+    r (float): how many examples to use for validation, in the range from 0 to 1
+
+    """
+    val_path = os.path.join(os.path.split(path)[0], 'val')
```

Comment:
In `ImageClassifierData.from_paths` in this file we have a default `val_name='valid'` (and that's the convention used in the lessons), would it make sense to use the same default here? Perhaps `create_val` could take the same parameter with the same default so the methods closely mirror each other? :)

BTW, nice idea!

Reply:
Good thinking! Changes made